src/processes/helper/addUserToTopicChannel.js

```diff
@@ -1,41 +1,39 @@
-const { convertString, ICON_TOPIC_CHANNEL, CHANNEL_TYPE_TOPIC } = require('../../utils');
-const { LogError } = require('../../databases/models');
-const StreamChat = require("stream-chat").StreamChat;
+const {convertString, ICON_TOPIC_CHANNEL, CHANNEL_TYPE_TOPIC} = require('../../utils');
+const {LogError} = require('../../databases/models');
+const StreamChat = require('stream-chat').StreamChat;
 
 const addUserToTopicChannel = async (user_id, topics) => {
-    try {
-        let userId = user_id.toString();
-        const serverClient = StreamChat.getInstance(
-            process.env.API_KEY,
-            process.env.SECRET
-        );
-        topics.map(async (item) => {
-            serverClient.createToken(userId);
-            const channelId = item;
-            const members = [];
-            members.push(userId);
+  try {
+    let userId = user_id.toString();
+    const serverClient = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+    topics.map(async (topic) => {
+      const newTopic = convertString(topic, '-', '').toLowerCase();
+      const token = serverClient.createToken(userId);
 
-            let channelName = "#" + convertString(item, "-", "");
+      await serverClient.connectUser({id: userId}, token);
+      const members = [];
+      members.push(userId);
 
-            const channel = serverClient.channel("topics", channelId, {
-                name: channelName,
-                created_by_id: "system",
-                channel_type: CHANNEL_TYPE_TOPIC,
-                channelImage: ICON_TOPIC_CHANNEL,
-                channel_image: ICON_TOPIC_CHANNEL,
-                image: ICON_TOPIC_CHANNEL
-            });
-            await channel.create();
+      const channel = serverClient.channel('topics', `topic_${newTopic}`, {
+        name: `#${newTopic}`,
+        channel_type: CHANNEL_TYPE_TOPIC,
+        channelImage: ICON_TOPIC_CHANNEL,
+        channel_image: ICON_TOPIC_CHANNEL,
+        image: ICON_TOPIC_CHANNEL
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are three properties (`channelImage`, `channel_image`, `image`) set to the same value (`ICON_TOPIC_CHANNEL`). If these properties serve the same purpose, consider keeping only one to avoid redundancy.

```diff
-        channelImage: ICON_TOPIC_CHANNEL,
-        channel_image: ICON_TOPIC_CHANNEL,
-        image: ICON_TOPIC_CHANNEL
+        image: ICON_TOPIC_CHANNEL
```

27:
The `sendMessage` method is called with `{skip_push: true}` as the second argument, but according to the StreamChat documentation, this method only takes one argument. Please verify if this is an error or if the library has been customized to accept a second argument.

```diff
-      await channel.sendMessage({text: 'This topic has new followers'}, {skip_push: true});
+      await channel.sendMessage({text: 'This topic has new followers'});
```

38:
There's no newline at the end of the file. While this isn't a critical issue, it's a common convention to end files with a newline because some tools require it and it can prevent potential issues in the future.

```diff
- module.exports = addUserToTopicChannel;
+ module.exports = addUserToTopicChannel;\n
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->